adapters/akcelo/akcelo.go

```diff
@@ -0,0 +1,188 @@
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	requestData, errs := a.prepareBidRequest(request)
+	return requestData, errs
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(response) {
+		return nil, nil
+	}
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{err}
+	}
+	var bidResponse openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(response.Body, &bidResponse); err != nil {
+		return nil, []error{err}
+	}
+	return extractBids(bidResponse)
+}
+
+func (a *adapter) prepareBidRequest(bidRequest *openrtb2.BidRequest) ([]*adapters.RequestData, []error) {
+	bidRequest = createSitePublisher(bidRequest)
+	requestData := make([]*adapters.RequestData, 0, 1)
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	var errs []error
+	for i, imp := range bidRequest.Imp {
+		newImp, err := a.prepareImp(imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if newImp != nil {
+			bidRequest.Imp[i] = *newImp
+		}
+	}
+
+	if len(bidRequest.Imp) > 0 {
+		if err := configureParentAccount(bidRequest); err != nil {
+			return nil, []error{err}
+		}
+	} else {
+		return nil, []error{&errortypes.BadInput{Message: fmt.Sprintf("No valid Imp")}}
+	}
+
+	var resultBidRequest, err = jsonutil.Marshal(bidRequest)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	requestData = append(requestData, &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.uri.String(),
+		Body:    resultBidRequest,
+		Headers: headers,
+		ImpIDs:  openrtb_ext.GetImpIDs(bidRequest.Imp),
+	})
+	return requestData, errs
+}
+
+func createSitePublisher(bidRequest *openrtb2.BidRequest) *openrtb2.BidRequest {
+	akceloRequest := *bidRequest
+	if akceloRequest.Site == nil {
+		akceloRequest.Site = &openrtb2.Site{}
+	} else {
+		siteCopy := *akceloRequest.Site
+		akceloRequest.Site = &siteCopy
+	}
+	if akceloRequest.Site.Publisher == nil {
+		akceloRequest.Site.Publisher = &openrtb2.Publisher{}
+	} else {
+		publisherCopy := *akceloRequest.Site.Publisher
+		akceloRequest.Site.Publisher = &publisherCopy
+	}
+	return &akceloRequest
+}
+
+func configureParentAccount(bidRequest *openrtb2.BidRequest) error {
+	parentAccount, _, _, err := jsonparser.Get(bidRequest.Imp[0].Ext, "akcelo", "siteId")
+	if err != nil {
+		return &errortypes.BadInput{Message: fmt.Sprintf("Cannot find valid siteId")}
+	}
+	var publisherExt = openrtb_ext.ExtPublisher{}
+	publisherExt.Prebid = &openrtb_ext.ExtPublisherPrebid{}
+	var parentAccountStr = string(parentAccount)
+	publisherExt.Prebid.ParentAccount = &parentAccountStr
+	bidRequest.Site.Publisher.Ext, _ = jsonutil.Marshal(&publisherExt)
```

Comment:
@rogerDyl : why are we suppressing the error here? and not returning it outside?